load_confounds/parser.py

```diff
@@ -332,11 +334,18 @@ def _load_single(self, confounds_raw):
             )
             confounds = pd.concat([confounds, confounds_compcor], axis=1)
 
+
+        # Convert from DataFrame to numpy ndarray
         labels = confounds.columns
+        confounds = confounds.values
+
+        # Derivatives have NaN
+        # Replace them by zeros, otherwise nilearn will not regress them out.
+        confounds[np.isnan(confounds)] = 0
```

Comment:
confounds = np.nan_to_num(confounds)

Reply:
your call it won't change the output

Reply:
`np.nan_to_num(confounds, copy=False)` 
more efficient

Reply:
or
`confounds = np.nan_to_num(confounds, nan=np.mean(confounds))`

Reply:
not sure the mean of all confounds would make sense, this would had to be column by column

Reply:
thanks for the tip.

Reply:
In the current version NaNs are not replaced by zeros anymore. See the thread below.